File: server/posts/models/comments.go
package models

import "time"

// Comment model
type Comment struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	UserID    uint      `json:"creatorId"`
	PostID    uint      `json:"postId"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	Body      string    `json:"body"`
	Creator   string    `json:"creator" gorm:"->"`
}

// model for comments pagination
type PaginatedComments struct {
	Comments []Comment `json:"comments"`
	HasMore  bool      `json:"hasMore"`
}

File: web/src/requests/comments.ts
export interface Comment {
  creatorId: number;
  postId: number;
  createdAt: string;
  updatedAt: string;
  body: string;
  creator: string;
}

export interface PaginatedComments {
  comments: Comment[];
  hasMore: boolean;
}

const url = process.env.API_URL;
const headers = {
  Accept: "application/json",
  "Content-Type": "application/json",
};

export const addComment = async (
  id: number,
  body: string
): Promise<Comment> => {
  const res = await fetch(`${url}/posts/${id}/comments`, {
    method: "POST",
    body: JSON.stringify({
      body: body,
    }),
    headers: headers,
    credentials: "include",
  });

  if (!res.ok) {
    const err = await res.json();
    throw new Error(err)
  }

  return await res.json();
};

export const deleteComment = async (id: number) => {
  const res = await fetch(`${url}/posts/${id}/comments`, {
    method: "DELETE",
    headers: headers,
    credentials: "include"
  });

  if (!res.ok) {
    const err = await res.json();
    throw new Error(err)
  }

  return await res.json();
}
